src/main/java/edu/cornell/kfs/module/purap/document/validation/IWantNoteRule.java

```diff
@@ -0,0 +1,30 @@
+package edu.cornell.kfs.module.purap.document.validation;
+
+import org.kuali.kfs.kns.document.MaintenanceDocument;
+import org.kuali.kfs.kns.maintenance.rules.MaintenanceDocumentRuleBase;
+import org.kuali.kfs.krad.service.SequenceAccessorService;
+import org.kuali.kfs.krad.util.ObjectUtils;
+import org.kuali.kfs.sys.context.SpringContext;
+
+import edu.cornell.kfs.module.purap.CUPurapConstants;
+import edu.cornell.kfs.module.purap.businessobject.IWantNote;
+
+public class IWantNoteRule extends MaintenanceDocumentRuleBase {
+
+    @Override
+    protected boolean processCustomRouteDocumentBusinessRules(MaintenanceDocument document) {
+        boolean isValid = super.processCustomRouteDocumentBusinessRules(document);
+        if (isValid) {
+            final IWantNote iWantNote = (IWantNote) document.getNewMaintainableObject().getBusinessObject();
+            if (ObjectUtils.isNull(iWantNote.getNoteIdentifier())) {
+                final SequenceAccessorService sequenceAccessorService = SpringContext
```

Comment:
Is SequenceAccessorService needed?  If the note Identifier isn't set, the business object service should get the next sequence from OJB.

Reply:
Great question. The issue I was trying to solve is the fact that the maintenance doc will not have the note identifier if the value is set through the OJB. The new BO will have the identifier set and saved in the DB but the Doc does not have the value. I tried a couple other ways to try to force the value on the doc but this was the only way I got it to work (and the only example I found in the code base). If you have another idea I could try please let me know. Thank you!

Reply:
Thanks for the explanation, I think this solution makes sense.